### PR TITLE
Makes it an error to store during assembly of a call

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Before this, there was some extra code in the throttle package handling
a bug in our in memory storage. This fixes that and removes the extra
code.

See #2502

### CODE REVIEW

File: zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledCall.java [hunk 1]
```diff
@@ -41,39 +39,21 @@
...
-   * supplier call needs to be supplied later to avoid having it take action when it is created
-   * (like {@link InMemoryStorage} and thus avoid being throttled.
-   */
-  final Supplier<? extends Call<V>> supplier;
-  volatile Call<V> delegate;
-  volatile boolean canceled;
-
-  public ThrottledCall(ExecutorService executor, Limiter<Void> limiter,
-    Supplier<? extends Call<V>> supplier) {
+  final Call<V> delegate;
```
Comment (codefromthecrypt) [inline zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledCall.java:45]: look ma. no volatiles :)

**Comment by codefromthecrypt** [comment]
finally Jenkins is green

File: zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledCall.java [hunk 2]
```diff
@@ -153,17 +126,17 @@ static String setCurrentThreadName(String name) {
 
   final class QueuedCall implements Runnable {
```
Comment (Logic-32) [inline zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledCall.java]: Can this be `static` now?

File: zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledCall.java (hunk 1 above)
Comment (Logic-32) [reply zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledCall.java:45]: :D

File: zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledCall.java (hunk 2 above)
Comment (codefromthecrypt) [reply zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledCall.java]: yep fixing in the pending change thx

**Pull request merged**